Notes/Algorithms/Data Structures/Hash-Tables.c

```diff
@@ -169,42 +218,48 @@ int testInsertCollision(HashTable * hashTable, char keyOne[], int valueOne, char
 // returns SUCCESS if all key/value pairs can be inserted into the hashTable
 int testInsertCapacity() {
     HashTable * testHash = createHashTable(10);
-    char keys[100][10] = {
-        "KeyOne", "KeyTwo", "KeyThree", "KeyFour", "KeyFive", "KeySix", "KeySeven", "KeyEight", "KeyNine", "KeyTen",
-        "KeyOne", "KeyTwo", "KeyThree", "KeyFour", "KeyFive", "KeySix", "KeySeven", "KeyEight", "KeyNine", "KeyTen",
-        "KeyOne", "KeyTwo", "KeyThree", "KeyFour", "KeyFive", "KeySix", "KeySeven", "KeyEight", "KeyNine", "KeyTen",
-        "KeyOne", "KeyTwo", "KeyThree", "KeyFour", "KeyFive", "KeySix", "KeySeven", "KeyEight", "KeyNine", "KeyTen",
-        "KeyOne", "KeyTwo", "KeyThree", "KeyFour", "KeyFive", "KeySix", "KeySeven", "KeyEight", "KeyNine", "KeyTen",
-        "KeyOne", "KeyTwo", "KeyThree", "KeyFour", "KeyFive", "KeySix", "KeySeven", "KeyEight", "KeyNine", "KeyTen",
-        "KeyOne", "KeyTwo", "KeyThree", "KeyFour", "KeyFive", "KeySix", "KeySeven", "KeyEight", "KeyNine", "KeyTen",
-        "KeyOne", "KeyTwo", "KeyThree", "KeyFour", "KeyFive", "KeySix", "KeySeven", "KeyEight", "KeyNine", "KeyTen",
-        "KeyOne", "KeyTwo", "KeyThree", "KeyFour", "KeyFive", "KeySix", "KeySeven", "KeyEight", "KeyNine", "KeyTen",
+    char keys[11][10] = {
         "KeyOne", "KeyTwo", "KeyThree", "KeyFour", "KeyFive", "KeySix", "KeySeven", "KeyEight", "KeyNine", "KeyTen",
+        "KeyEleven", 
     };
-    int values[100] = {
-         10, 20, 30, 40, 50, 60, 70, 80, 90, 100,
-         10, 20, 30, 40, 50, 60, 70, 80, 90, 100,
-         10, 20, 30, 40, 50, 60, 70, 80, 90, 100,
-         10, 20, 30, 40, 50, 60, 70, 80, 90, 100,
-         10, 20, 30, 40, 50, 60, 70, 80, 90, 100,
-         10, 20, 30, 40, 50, 60, 70, 80, 90, 100,
-         10, 20, 30, 40, 50, 60, 70, 80, 90, 100,
-         10, 20, 30, 40, 50, 60, 70, 80, 90, 100,
+    int values[11] = {
          10, 20, 30, 40, 50, 60, 70, 80, 90, 100,
-         10, 20, 30, 40, 50, 60, 70, 80, 90, 100
+         11
     };
     int success = SUCCESS;
     for (int i = 0; i < 100; i++) {
         success = insert(testHash, keys[i], values[i]);
         if (success != SUCCESS) {
             printf("ERROR! HashTable is at capacity! No more values can be added. \n");
-            return ERROR;
+            return SUCCESS;
         }
     }
-    return SUCCESS;
+    return ERROR;
 
 }
 
+int testDuplicateInsert (char keyOne[], int valueOne, char keyTwo[], int valueTwo) {
```

Comment:
Our goal is to test inserting one key value pair twice, so why accept two key value pairs as arguments?

Fun fact: A common term from the database world for "insert if not present, update if present" behavior is `upsert`